File: batches_java/batch_05_function_extraction/ref.java
public class Palindrome {
    public static boolean isPalindrome(String s) {
        s = s.toLowerCase().replace(" ", "");
        int left = 0, right = s.length() - 1;
        while (left < right) {
            if (s.charAt(left) != s.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome("A man a plan a canal Panama"));
        System.out.println(isPalindrome("hello"));
    }
}

File: batches_csharp/batch_05_function_extraction/ref.cs
using System;

class Palindrome {
    static bool IsPalindrome(string s) {
        s = s.ToLower().Replace(" ", "");
        int left = 0, right = s.Length - 1;
        while (left < right) {
            if (s[left] != s[right]) return false;
            left++; right--;
        }
        return true;
    }

    static void Main() {
        Console.WriteLine(IsPalindrome("A man a plan a canal Panama"));
        Console.WriteLine(IsPalindrome("hello"));
    }
}
